torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -5301,6 +5301,42 @@ def test_custom_stream_nested_multi(self):
             {"cuda:0": "cuda:1", "cuda:1": "cuda:0"}
         )
 
+    @staticmethod
+    def _gpu_add_wrong_gpus(x, y):
+        if x.is_cuda and y.is_cuda:
+            return x.cpu() + y.cuda()
```

Comment:
Do we need to have an RPC that errors out to test the code changes? Shouldn't a regular RPC uncover the bug and make this test fail?

Reply:
regular RPC and cuda RPC go through different code path. Regular RPC will return an `ivalue::Future`, which does not hit the subvalues problem. The CUDA RPC returns a `CUDAFuture`, and this is where we hit the error. It is true that we don't need a new static method here, but I just wanna make sure that common wrong device error on remote side can be correctly propagated to the caller, as this might be an common error when people start to use CUDA RPC.